pkg/pgmodel/querier/query_builder.go

```diff
@@ -216,68 +218,107 @@ type aggregators struct {
 	tsSeries    TimestampSeries //can be NULL and only present if timeClause == ""
 }
 
-/* The path is the list of ancestors (direct parent last) returned node is the most-ancestral node processed by the pushdown */
-// todo: investigate if query hints can have only node and lookback
-func getAggregators(md *promqlMetadata) (*aggregators, parser.Node, error) {
-	path := md.path // PromQL AST.
-	qh := md.queryHints
-	hints := md.selectHints
-	if !extension.ExtensionIsInstalled || qh == nil || hasSubquery(path) || hints == nil {
-		return getDefaultAggregators(), nil, nil
+// getAggregators returns the aggregator which should be used to fetch data for
+// a single metric. It may apply pushdowns to functions.
+func getAggregators(metadata *promqlMetadata) (*aggregators, parser.Node, error) {
```

Comment:
I actually think that `getAggregators` should be moved over into `query_builder_samples.go`, but I didn't want to mix that move with this small refactoring.